eng/tools/sdk-suppressions/src/updateSdkSuppressionsLabel.ts

```diff
@@ -0,0 +1,335 @@
+import _ from "lodash";
+import { writeFileSync } from "fs";
+import { Octokit, RestEndpointMethodTypes } from "@octokit/rest";
+import { OctokitResponse, ReposGetContentResponseData } from "@octokit/types";
+import { sdkLabels, SdkName } from './sdk.js';
+import { 
+    SdkSuppressionsYml, 
+    SdkSuppressionsSection, 
+    sdkSuppressionsFileName, 
+    SdkPackageSuppressionsEntry, 
+    validateSdkSuppressionsFile 
+} from './sdkSuppressions.js';
+import {
+  base64ToString,
+  parseYamlContent,
+} from "./common.js";
+
+export type PullRequestContext = {
+    labels: string[];
+    number: number;
+    html_url: string;
+    head: {
+        owner: string,
+        repo: string,
+        ref: string
+    };
+    base: {
+        owner: string,
+        repo: string,
+        ref: string
+    };
+}
+
+/**
+ * 
+ * @param appClient 
+ * @param pr 
+ * @returns SdkName list
+ * This part compares the suppression files of the head branch and the base branch. 
+ * To get the SDK, we need to identify which package name, SDK name, or breaking changes are different and apply the SDK suppression label accordingly in the next step.
+ * change details can see at getSdkNamesWithChangedSuppressions function
+ * on the other hand that the sdkName list will return an empty array if it does not have a suppression file or if the file is blank.
+ */
+export async function getSdkSuppressionsSdkNames(
+  appClient: Octokit,
+  pr: PullRequestContext
+): Promise<SdkName[]> {
+  // Get changed files paths
+  const filesChangedPaths: string[] = await listChangedFilePaths(appClient, {
```

Comment:
Could we invoke the existed common function in powershell to get the changed files then pass in as the context?

Reply:
added